features/consent/src/main/java/in/projecteka/jataayu/consent/ui/fragment/ConsentsListFragment.kt

```diff
@@ -80,6 +87,54 @@ abstract class ConsentsListFragment : BaseFragment(), AdapterView.OnItemSelected
                 }
             }
         })
+
+        if (getConsentFlow() == ConsentFlow.GRANTED_CONSENTS) {
```

Comment:
no need to do this check. create a method initObservers() to register all your observers, based on business logic these observers will get triggered if there is an event emitted.